tests/e2e/multichain/utils.go

```diff
@@ -0,0 +1,276 @@
+			CoinType:         "60",
+			GasPrices:        minGasPrice + denom,
+			GasAdjustment:    1.5,
+			TrustingPeriod:   "168h0m0s",
+			ModifyGenesis:    cosmos.ModifyGenesis(genesis),
+			CoinDecimals:     &decimals,
+			SigningAlgorithm: "eth_secp256k1",
+			// open the port for the EVM on all nodes
+			ExposeAdditionalPorts: []string{"8545/tcp"},
+			EncodingConfig:        encoding,
+		},
+	}
+}
+
+func StartXplaChain(t *testing.T, ctx context.Context, version []ibc.DockerImage) (*cosmos.CosmosChain, *client.Client) {
+	config := sdk.GetConfig()
+	config.SetBech32PrefixForAccount("xpla", "xplapub")
+
+	rep := testreporter.NewNopReporter()
+	eRep := rep.RelayerExecReporter(t)
+
+	client, network := interchaintest.DockerSetup(t)
+
+	cf := interchaintest.NewBuiltinChainFactory(zaptest.NewLogger(t), []*interchaintest.ChainSpec{
+		XplaChainSpec(1, 0, "interchaintest_1-1", version),
+	})
+	chains, err := cf.Chains(t.Name())
+	assert.NoError(t, err)
+
+	chain := chains[0].(*cosmos.CosmosChain)
+
+	ic := interchaintest.NewInterchain().AddChain(chain)
+	err = ic.Build(ctx, eRep, interchaintest.InterchainBuildOptions{
+		TestName:         t.Name(),
+		Client:           client,
+		NetworkID:        network,
+		SkipPathCreation: true,
+	})
+	assert.NoError(t, err)
+
+	t.Cleanup(func() {
+		_ = ic.Close()
+	})
+
+	return chain, client
+}
+
+type IBCChainSetup struct {
+	XplaChain   *cosmos.CosmosChain
+	SimdChain   *cosmos.CosmosChain
+	XplaUsers   []ibc.Wallet
+	SimdUsers   []ibc.Wallet
+	Client      *client.Client
+	chains      []ibc.Chain
+	r           ibc.Relayer
+	ibcPathName string
+	eRep        *testreporter.RelayerExecReporter
+}
+
+func (s *IBCChainSetup) GetSimdChannels(ctx context.Context) ([]ibc.ChannelOutput, error) {
+	return s.r.GetChannels(ctx, s.eRep, s.SimdChain.Config().ChainID)
+}
+
+func (s *IBCChainSetup) FlushRelayer(ctx context.Context) error {
+	xplaChannels, err := s.r.GetChannels(ctx, s.eRep, s.XplaChain.Config().ChainID)
+	if err != nil {
+		return err
+	}
+	for _, channel := range xplaChannels {
+		if err := s.r.Flush(ctx, s.eRep, s.ibcPathName, channel.ChannelID); err != nil {
+			return err
+		}
+	}
+
+	simdChannels, err := s.r.GetChannels(ctx, s.eRep, s.SimdChain.Config().ChainID)
+	if err != nil {
+		return err
+	}
+	for _, channel := range simdChannels {
+		if err := s.r.Flush(ctx, s.eRep, s.ibcPathName, channel.ChannelID); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *IBCChainSetup) CreateChannel(ctx context.Context, srcPortName, destPortName, version string) error {
+	return s.r.CreateChannel(ctx, s.eRep, s.ibcPathName, ibc.CreateChannelOptions{
+		SourcePortName: srcPortName,
+		DestPortName:   destPortName,
+		Order:          ibc.Unordered,
+		Version:        version,
+		Override:       true,
+	})
+}
+
+func StartXplaChainAndSimdWithIBC(t *testing.T, ctx context.Context, version []ibc.DockerImage) *IBCChainSetup {
+	config := sdk.GetConfig()
+	config.SetBech32PrefixForAccount("xpla", "xplapub")
```

Comment:
Same here. To set config once, it would be better move config getting and setting codes to init(). I think this could cause the race condition when concurrent testing.